File: Branches/VBInicial/vpPriV100GrupoMundifios/FornecedoresCertificados/Base/FichaFornecedor/BasIsFichaFornecedor.vb
'Imports Primavera.Extensibility.Base.Editors
'Imports Primavera.Extensibility.BusinessEntities.ExtensibilityService
'Imports Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs
'Imports Primavera.Extensibility.Constants.ExtensibilityService
'Imports StdBE100
'Imports Vimaponto.PrimaveraV100.Clientes.GrupoMundifios.Generico


'Namespace FornecedoresCertificados
'    Public Class BasIsFichaFornecedor
'        Inherits FichaFornecedores

'        Public Overrides Sub TeclaPressionada(KeyCode As Integer, Shift As Integer, e As ExtensibilityEventArgs)
'            MyBase.TeclaPressionada(KeyCode, Shift, e)

'            If Module1.VerificaToken("FornecedoresCertificados") = 1 Then
'                '    
'                'Crtl + Q JFC 04/11/2019
'                If KeyCode = 81 And Me.Fornecedor.Inactivo = False Then
'                    Module1.certEntidade = Me.Fornecedor.Fornecedor


'                    Dim result As ExtensibilityResult = PriV100Api.BSO.Extensibility.CreateCustomFormInstance(GetType(FrmFornecedoresCertsViews))

'                    If result.ResultCode = ExtensibilityResultCode.Ok Then

'                        Dim frm As FrmFornecedoresCertsViews = result.Result
'                        frm.ShowDialog()

'                    End If

'                End If

'                If KeyCode = 81 And Me.Fornecedor.Inactivo = True Then
'                    MsgBox("Fornecedor Anulado! Não é possível abrir o formulário de certificados!", vbInformation)
'                End If

'            End If
'        End Sub

'    End Class
'End Namespace

File: Trunk/vpPriV100GrupoMundifios/FornecedoresCertificados/Base/FichaFornecedor/BasIsFichaFornecedor.cs
using Generico;
using Primavera.Extensibility.Base.Editors;
using Primavera.Extensibility.BusinessEntities.ExtensibilityService;
using Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs;
using Primavera.Extensibility.Constants.ExtensibilityService;
using System.Windows.Forms;

namespace FornecedoresCertificados
{
    public class BasIsFichaFornecedor : FichaFornecedores
    {
        public override void TeclaPressionada(int KeyCode, int Shift, ExtensibilityEventArgs e)
        {
            base.TeclaPressionada(KeyCode, Shift, e);

            if (Module1.VerificaToken("FornecedoresCertificados") == 1)
            {
                //
                // Crtl + R JFC 04/11/2019
                if (KeyCode == 82 & this.Fornecedor.Inactivo == false)
                {
                    Module1.certEntidade = this.Fornecedor.Fornecedor;

                    ExtensibilityResult result = BSO.Extensibility.CreateCustomFormInstance(typeof(FrmFornecedoresCertsView));

                    if (result.ResultCode == ExtensibilityResultCode.Ok)
                    {
                        FrmFornecedoresCertsView frm = result.Result;
                        frm.ShowDialog();
                    }
                }

                if (KeyCode == 81 & this.Fornecedor.Inactivo == true)
                    MessageBox.Show("Fornecedor Anulado! Não é possível abrir o formulário de certificados!", "", MessageBoxButtons.OK, MessageBoxIcon.Information);
            }
        }
    }
}